src/State.ts

```diff
@@ -180,17 +174,8 @@ export class State<TContext, TEvent extends EventObject = EventObject>
     });
     this.matches = this.matches.bind(this);
     this.toStrings = this.toStrings.bind(this);
-  }
-
-  /**
-   * The next events that will cause a transition from the current state.
-   */
-  public get nextEvents(): EventType[] {
-    if (!this.tree) {
-      return [];
-    }
 
-    return this.tree.nextEvents;
+    this.nextEvents = !config.tree ? [] : config.tree.nextEvents;
```

Comment:
This is a potentially expensive operation.

Reply:
Should I change it to a regular method then? Was thinking about removing it entirely, but that would force consumers to handle missing tree themselves, which is not that ergonomic.

Reply:
Let's change it to a method, then. It's not public API.